src/methods/publishPost.ts

```diff
@@ -42,29 +42,146 @@ export const publishPost = async (
 	const data = matter(view.getViewData());
 
 	const frontmatter = {
+		id: metaMatter?.id || undefined,
```

Comment:
The existing code is providing default values in case the applicable note property is unset, but a default value isn't needed in this case. Your code is essentially saying "make `id` the value of `metaMatter.id?`, unless it's nothing, in which case make `id` nothing", when just `metaMatter?.id` should have the same effect. The same goes for Line 47
```suggestion
		id: metaMatter?.id,
```